File: cases/taint/cpp/computed-property-positive/computed_property_positive.cpp
#include <map>
#include <string>

std::string dfb_source() { // DFB-SOURCE: computed-property-input
    return "tainted";
}

void dfb_sink(const std::string &value) {} // DFB-SINK: computed-property-sink

void run() {
    std::map<std::string, std::string> holder;
    std::string key = "payload";
    holder[key] = dfb_source(); // DFB-WITNESS: computed-property-store
    dfb_sink(holder[key]);
}

File: cases/taint/go/computed-property-positive/computed_property_positive.go
package dataflowbench

import "reflect"

func dfb_source() string { // DFB-SOURCE: computed-property-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: computed-property-sink

type Holder struct {
	Payload string
	Other   string
}

func run() {
	key := "Payload"
	holder := Holder{Payload: "clean", Other: "clean"}
	fields := reflect.ValueOf(&holder).Elem()
	fields.FieldByName(key).SetString(dfb_source()) // DFB-WITNESS: computed-property-store
	dfb_sink(fields.FieldByName(key).String())
}
